lib/src/kvs/tx.rs

```diff
@@ -649,50 +674,71 @@ impl Transaction {
 	///
 	/// This function fetches the full range of key-value pairs, in a single request to the underlying datastore.
 	#[allow(unused_variables)]
-	pub async fn scan<K>(&mut self, rng: Range<K>, limit: u32) -> Result<Vec<(Key, Val)>, Error>
+	pub async fn scan<K>(
+		&mut self,
+		page: ScanPage<K>,
+		batch_limit: u32,
+	) -> Result<ScanResult<K>, Error>
```

Comment:
Could we make this function `pub async fn range<K>()`? And keep the old `scan()` function. For code which always wants to get a set of keys (without the paging and iteration), that will always be more efficient, because of the reduced key cloning.

Reply:
I would indeed keep the current API, maybe add a paged_scan() methods.
In terms of code design, I would be more comfortable if the method would return an iterator. Then the iterator can itself handle paging. That would be transparent for the caller, and would avoid asking the user to handle complex types (ScanPage). WDYT?

Reply:
Reverted original function, new function is `scan_paged`. Will go over the invocations and revert necessary ones as well